[CI] Internal data frame indices can cause unrelated docs tests to fail

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+intake/84/console failed due to the `.data-frame-notifications-1` index existing in a docs test that didn't expect it to exist:

```
org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT > test {yaml=reference/api-conventions/line_345} FAILED
    java.lang.AssertionError: Failure at [reference/api-conventions:275]: $body didn't match expected value:
                             $body: 
                            metadata: 
                               indices: 
                                 index-1: 
                                     state: same [open]
                                 index-2: 
                                     state: same [open]
                                 index-3: 
                                     state: same [open]
             .data-frame-notifications-1: unexpected but found [{state=open}]

        Caused by:
        java.lang.AssertionError: $body didn't match expected value:
                                 $body: 
                                metadata: 
                                   indices: 
                                     index-1: 
                                         state: same [open]
                                     index-2: 
                                         state: same [open]
                                     index-3: 
                                         state: same [open]
                 .data-frame-notifications-1: unexpected but found [{state=open}]
```

I imagine we had similar problems previously with ML internal indices leaking into unrelated docs tests.  We should look at how that was solved and implement something similar for data frame internal indices.

--- a/docs/reference/data-frames/apis/put-transform.asciidoc
+++ b/docs/reference/data-frames/apis/put-transform.asciidoc
@@ -110,7 +110,7 @@ PUT _data_frame/transforms/ecommerce_transform
 }
 --------------------------------------------------
 // CONSOLE
-// TEST[skip: https://github.com/elastic/elasticsearch/issues/43271]
+// TEST[setup:kibana_sample_data_ecommerce]
 
 When the transform is created, you receive the following results:
 [source,js]

--- a/docs/reference/ml/apis/put-job.asciidoc
+++ b/docs/reference/ml/apis/put-job.asciidoc
@@ -116,7 +116,6 @@ PUT _ml/anomaly_detectors/total-requests
 }
 --------------------------------------------------
 // CONSOLE
-// TEST[skip: https://github.com/elastic/elasticsearch/issues/43271]
 
 When the job is created, you receive the following results:
 [source,js]

--- a/docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java
+++ b/docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java
@@ -33,6 +33,7 @@ import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.common.xcontent.XContentLocation;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.common.xcontent.XContentParser.Token;
+import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.test.rest.yaml.ClientYamlDocsTestClient;
 import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
 import org.elasticsearch.test.rest.yaml.ClientYamlTestClient;
@@ -41,6 +42,7 @@ import org.elasticsearch.test.rest.yaml.ClientYamlTestResponse;
 import org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase;
 import org.elasticsearch.test.rest.yaml.restspec.ClientYamlSuiteRestSpec;
 import org.elasticsearch.test.rest.yaml.section.ExecutableSection;
+import org.junit.After;
 
 import java.io.IOException;
 import java.util.ArrayList;
@@ -95,6 +97,24 @@ public class DocsClientYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
             final Version esVersion,
             final Version masterVersion) {
         return new ClientYamlDocsTestClient(restSpec, restClient, hosts, esVersion, masterVersion, this::getClientBuilderWithSniffedHosts);
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        if (isMachineLearningTest() || isDataFrameTest()) {
+            logger.info("{}: waiting for pending tasks");
+            ESRestTestCase.waitForPendingTasks(adminClient());
+        }
+    }
+
+    protected boolean isMachineLearningTest() {
+        String testName = getTestName();
+        return testName != null && (testName.contains("=ml/") || testName.contains("=ml\\"));
+    }
+
+    protected boolean isDataFrameTest() {
+        String testName = getTestName();
+        return testName != null && (testName.contains("=data_frames/") || testName.contains("=data_frames\\"));
     }
 
     /**

--- a/docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java
+++ b/docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java
@@ -102,19 +102,18 @@ public class DocsClientYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
     @After
     public void cleanup() throws Exception {
         if (isMachineLearningTest() || isDataFrameTest()) {
-            logger.info("{}: waiting for pending tasks");
             ESRestTestCase.waitForPendingTasks(adminClient());
         }
     }
 
     protected boolean isMachineLearningTest() {
         String testName = getTestName();
-        return testName != null && (testName.contains("=ml/") || testName.contains("=ml\\"));
+        return testName != null && (testName.contains("/ml/") || testName.contains("\\ml\\"));
     }
 
     protected boolean isDataFrameTest() {
         String testName = getTestName();
-        return testName != null && (testName.contains("=data_frames/") || testName.contains("=data_frames\\"));
+        return testName != null && (testName.contains("/data-frames/") || testName.contains("\\data-frames\\"));
     }
 
     /**

